pkg/core/dao/dao.go

```diff
@@ -180,21 +180,75 @@ func (dao *Simple) putTokenTransferInfo(acc util.Uint160, bs *state.TokenTransfe
 
 // -- start transfer log.
 
-func getTokenTransferLogKey(acc util.Uint160, index uint32, isNEP11 bool) []byte {
-	key := make([]byte, 1+util.Uint160Size+4)
+func getTokenTransferLogPrefix(acc util.Uint160, isNEP11 bool) []byte {
+	buf := make([]byte, 1+util.Uint160Size)
 	if isNEP11 {
-		key[0] = byte(storage.STNEP11Transfers)
+		buf[0] = byte(storage.STNEP11Transfers)
 	} else {
-		key[0] = byte(storage.STNEP17Transfers)
+		buf[0] = byte(storage.STNEP17Transfers)
 	}
-	copy(key[1:], acc.BytesBE())
-	binary.LittleEndian.PutUint32(key[util.Uint160Size:], index)
+	copy(buf[1:], acc.BytesBE())
+	return buf
+}
+
+func getTokenTransferLogKey(acc util.Uint160, newestTimestamp uint64, index uint32, isNEP11 bool) []byte {
+	key := make([]byte, 1+util.Uint160Size+8+4)
+	prefix := getTokenTransferLogPrefix(acc, isNEP11)
+	copy(key, prefix)
+	binary.BigEndian.PutUint64(key[1+util.Uint160Size:], newestTimestamp)
+	binary.BigEndian.PutUint32(key[1+util.Uint160Size+8:], index)
 	return key
 }
 
+// SeekNEP17TransferLog executes f for each NEP-17 transfer in log starting from
+// the transfer with the newest timestamp up to the oldest transfer. It continues
+// iteration until false is returned from f. The last non-nil error is returned.
+func (dao *Simple) SeekNEP17TransferLog(acc util.Uint160, newestTimestamp uint64, f func(*state.NEP17Transfer) (bool, error)) error {
+	prefix := getTokenTransferLogPrefix(acc, false)
+	rngStart := make([]byte, 8)
+	binary.BigEndian.PutUint64(rngStart, newestTimestamp)
```

Comment:
And will just subslice the timestamp from the key here (avoiding `rngStart` allocation).